torch/testing/_internal/common_utils.py

```diff
@@ -2089,3 +2089,23 @@ def inner(self, *args, **kwargs):
                 fn(self, *args, **kwargs)
         return inner
     return decorator
+
+# This is a wrapper that wraps a test to run this test twice, one with
+# coalesced=True, another with coalesced=False for coalesced/uncoalesced sparse tensors.
+def coalescedonoff(f):
```

Comment:
This approach seems OK but I think the decorator can be simplified and made more future-proof by passing *args **kwargs to the test, and just passing `coalesced` as an additional kwarg-only argument to the test.

So a test could have a signature:

```
test_foo(self, device, dtype, *, coalesced)
```

For example, and the varargs would pass through as expected.

What would you think of that tweak?

Reply:
(This would also mean the checks for nargs could be removed.)

Reply:
Agree, this could be simplified in few lines of code using `*args` and `**kwargs`